### PR TITLE
Add test files and modify mineral_test.py

### PR DESCRIPTION
**Pull request by heidiaclayton**
Still needed: two more test sub-images. The current image has all "no data" pixels. A clean-up function would be nice to remove the files generated by the test.

Also, I can't seem to find the download link for the ENVI version of the USGS library... I have it locally but I can't find where I got it from.

### CODE REVIEW

**Comment by lewismc** [comment]
ack @heidiaclayton cleanup functions are always welcome :)
You can also add the test files to .gitignore such that they are not added accidentally at a later stage. Thank you for working diligently on these tests... the toolkit is really starting to come together.

**Comment by heidiaclayton** [comment]
@browtayl @lewismc Do we have any AVIRIS-NG images on the AWS machine?

**Comment by ghost** [comment]
@heidiaclayton we have [more than 1TB](https://groups.google.com/forum/#!topic/coal-capstone/P4Kx-lK5TFQ) worth of imagery on the server. there's a link on the [USGS library homepage](https://speclab.cr.usgs.gov/spectral.lib06/) to the convolved library in this ftp directory: `ftp://ftpext.cr.usgs.gov/pub/cr/co/denver/speclab/pub/spectral.library/splib06.library/Convolved.libraries/`. see my existing test code for test teardown functions.

**Comment by heidiaclayton** [comment]
The current status here is: I'm downloading an AVIRIS-NG image to get two sub-images to add to the test. Other than that, there's not much left except for maybe checking the metadata the mineral classifier wrote to the file. @browtayl , do you have an opinion on this?

**Comment by ghost** [comment]
@heidiaclayton this is looking really good (and it turns out i do get commit notifications lol). i'm not able to review the code right this minute, but i left most of my thoughts about the scope of the tests in issue https://github.com/capstone-coal/pycoal/issues/36 and pull https://github.com/capstone-coal/pycoal/pull/53.

i plan on updating the mineral classifier, so the implementation doesn't need to be perfect. simply doing a sanity check on these subimages will add a lot of value and possibly detect unknown problems with the code.

thanks for tackling this. as you probably found, it is not that complicated but it's somewhat tedious. :+1:

**Comment by heidiaclayton** [comment]
AVIRIS-NG test image added, plus a couple checks in the classified metadata.

File: pycoal/tests/mineral_test.py [hunk 1]
```diff
@@ -112,3 +112,61 @@ def test_toRGB_AVC():
...
+
+    # make sure library is being opened as such
+    assert isinstance(tst_cls.library, spectral.io.envi.SpectralLibrary)
+
+    for f in (test_classifyImage_testFilename_1, test_classifyImage_testFilename_2, test_classifyImage_testFilename_3):
+        tst_cls.classifyImage(f, f[:-4] + "_class.hdr")
+
+        classified = spectral.open_image(f[:-4] + "_class.hdr")
+
+        assert classified.metadata.get(u'description') == u'PyCOAL 0.2 mineral classified image.'
```
Comment (ghost) [inline pycoal/tests/mineral_test.py:167]: this could be `pycoal.version`

**Comment by ghost** [comment]
@heidiaclayton @lewismc this looks good to merge to me

File: pycoal/tests/mineral_test.py [hunk 2]
```diff
@@ -112,3 +112,61 @@ def test_toRGB_AVC():
...
+        assert classified.metadata.get(u'file type') == u'ENVI Classification'
+
+        # assert there are no invalid class numbers
+        for i in classified.asarray().flatten():
+            assert 0 <= i <= len(tst_cls.library.names) + 1
```
Comment (ghost) [inline pycoal/tests/mineral_test.py:172]: as we have discussed, an additional thing we could test is that the class of each pixel equals the value in some test expected test subimage.

**Comment by ghost** [comment]
I am closing this pull request because i have merged `mineral-test` into `development` per pull https://github.com/capstone-coal/pycoal/pull/73.

**Pull request closed without merging**